library/nsxt_rest.py

```diff
@@ -0,0 +1,279 @@
+          "unique_id": "0361313c-20f0-42ba-aa77-e090277a50ac"
+      }
+'''
+
+
+import os
+import json
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.vmware_nsxt import vmware_argument_spec
+from ansible.module_utils.urls import basic_auth_header, fetch_url
+
+class VMwareNSXTRest():
+    def __init__(self, module):
+        self.module = module
+        self.mgr_hostname = module.params["hostname"]
+        self.mgr_username = module.params["username"]
+        self.mgr_password = module.params["password"]
+        self.path = module.params["path"]
+        self.method = module.params["method"]
+        self.src = module.params["src"]
+        self.content = module.params["content"]
+
+        self.manager_url = "https://{}".format(self.mgr_hostname)
+        self.headers = { "authorization": basic_auth_header(self.mgr_username, self.mgr_password) }
+
+        if self.src:
+            if os.path.isfile(self.src):
+                try:
+                    with open(self.src, "r") as f:
+                        self.content = json.loads(f.read())
+                except Exception as err:
+                    self.module.fail_json(msg="src read error: %s" % err)
+            else:
+                self.module.fail_json(msg="cannot find/access src '%s'" % self.src)
+
+    def error_code_check(self, info):
+        status = info.get('status')
+        if status >= 400:
+            self.module.fail_json(msg="error_code: %s, error_message: %s"
+                                      % (status, json.loads(info.get('body'))['error_message']))
+
+        if status == -1:
+            self.module.fail_json(msg="error_code: %s, error_message: %s" % (status, info.get('msg')))
+
+    def get_method(self, ignore_errors=False):
+        try:
+            (resp, info) = fetch_url(self.module, self.manager_url + self.path, method="GET", headers=self.headers)
+        except Exception as err:
+            self.module.fail_json(msg="nsxt rest api request error: %s, error url: %s"
+                                      % (err, self.manager_url + self.path))
+
+        if not ignore_errors:
+            self.error_code_check(info)
+
+        if resp:
+            return json.loads(resp.read())
+        else:
+            return ""
+
+    def update_method(self, method, ignore_errors=False):
+        self.headers.update({"Accept": "application/json", "Content-Type": "application/json"})
+        try:
+            (resp, info) = fetch_url(self.module, self.manager_url + self.path, method=method.upper(),
+                                     headers=self.headers, data=json.dumps(self.content))
+        except Exception as err:
+            self.module.fail_json(msg="nsxt rest api request error: %s, error url: %s"
+                                      % (err, self.manager_url + self.path))
+
+        if not ignore_errors:
+            self.error_code_check(info)
+
+    def delete_method(self, ignore_errors=False):
+        try:
```

Comment:
lines 220-227 are similar in delete, get, update methods. perhaps you could extract them to a common function to remove redundancy

Reply:
I changed the method functions to operate_nsxt function and made it more straightforward :)